ush/jediinc2fv3.py

```diff
@@ -22,57 +20,174 @@
+
+        if name in vardict:
+            # The temperature increment is needed to compute the delz
+            # increment.  Either T or t may be used in jedi increment
+            # files.  Set tinc to the appropriate case.
+            if name in ['T', 't']:
+                tinc = name
+
+            x = ncout.createVariable(vardict[name], 'f4', dimsout)
+            if len(variable.dimensions) in [3, 4]:
+                ncout[vardict[name]][:] = ncin[name][0, ...]
+            else:
+                ncout[vardict[name]][:] = ncin[name][:]
+
+    # Populate increment and guess fields
+    #   Note:  increment and guess fields have same shape
+    #     ps_inc is (time, lat, lon), ps_ges is {time, grid_yt, grid_xt)
+    #     t_inc is (time, lev, lat, lon), t_ges is (time, pfull, grid_yt, grid_xt)
+
+    ps_ges = ncges.variables['pressfc'][:]
+    t_ges = ncges.variables['tmp'][:]
+    q_ges = ncges.variables['spfh'][:]
+
+    ps_inc = ncin.variables['ps'][:]
+    t_inc = ncin.variables[tinc][:]
+    q_inc = ncin.variables['sphum'][:]
+
+    # Compute analysis ps
+    ps_anl = np.zeros((nlats, nlons), float)
+    ps_anl = ps_ges + ps_inc
+
+    # Set constants and compute derived constants
+    grav = 9.80665
+    airmw = 28.965
+    h2omw = 18.015
+    runiv = 8314.47
+    rdry = runiv/airmw
+    rvap = runiv/h2omw
+    cpdry = 3.5*rdry
+    fv = (rvap/rdry)-1
+    rdog = rdry/grav
+    kappa = rdry/cpdry
+    kap1 = kappa+1.0
+    kapr = 1.0/kappa
+
+    # Get ak,bk from guess file
+    nc_attrs = ncges.ncattrs()
+    ak = ncges.getncattr('ak')
+    bk = ncges.getncattr('bk')
+
+    # Compute guess and analysis interface pressures
+    prsi_ges = np.zeros((nlevs+1, nlats, nlons), float)
+    prsi_anl = np.zeros((nlevs+1, nlats, nlons), float)
+    k = 0
+    while k < nlevs+1:
+        prsi_ges[k, :, :] = ak[k]+bk[k]*ps_ges[:, :]
+        prsi_anl[k, :, :] = ak[k]+bk[k]*ps_anl[:, :]
+        k = k + 1
+
+    # Compute pressure increment (delp_inc).  Compute
+    # guess and analysis layer pressures using Philips method
+    delp_inc = np.zeros((nlevs, nlats, nlons), float)
+    prsl_ges = np.zeros((nlevs, nlats, nlons), float)
+    prsl_anl = np.zeros((nlevs, nlats, nlons), float)
+    k = 0
+    while k < nlevs:
+        dbk = bk[k+1] - bk[k]
+        delp_inc[k, :, :] = ps_inc[:, :] * dbk
+        prsl_ges[k, :, :] = ((prsi_ges[k+1, :, :]**kap1 - prsi_ges[k, :, :]**kap1) / (kap1*(prsi_ges[k+1, :, :] - prsi_ges[k, :, :])))**kapr
+        prsl_anl[k, :, :] = ((prsi_anl[k+1, :, :]**kap1 - prsi_anl[k, :, :]**kap1) / (kap1*(prsi_anl[k+1, :, :] - prsi_anl[k, :, :])))**kapr
+        k = k + 1
+
+    # Write delp increment to output file
+    x = ncout.createVariable('delp_inc', 'f4', dimsout_inc)
+    ncout['delp_inc'][:] = delp_inc[:]
+
+    # Compute analysis temperature andl specific humidity
+    t_anl = np.zeros((nlevs, nlats, nlons), float)
+    q_anl = np.zeros((nlevs, nlats, nlons), float)
+    t_anl = np.add(t_ges, t_inc)
```

Comment:
why is this `np.add` but then the next line is just `q_ges + q_inc`?

Reply:
I don't recall why I use different approaches on adjacent lines.  I may have been experimenting with different ways to add and forgot to go back and clean up the code.  Since I use `+` elsewhere, I replaced `np.add(t_ges, t_inc)` with `t_ges + t_inc`.  Done at [0e30f31](https://github.com/NOAA-EMC/GDASApp/commit/0e30f3196bde18e3f4eb3f6fd9ba59ce146645aa)